src/components/utils/test/thread_test.cc

```diff
@@ -175,6 +247,175 @@ TEST_F(ThreadTest, StartOneThreadTwice_ExpectTheSameThreadStartedTwice) {
   EXPECT_TRUE(delegate_->check_value());
 }
 
+TEST(ThreadTest2, StartAndJoinForce_Success) {
```

Comment:
@pvvasilev, 
[Best practices in unit test writing](https://adc.luxoft.com/confluence/pages/viewpage.action?pageId=267764838)